allocator/Cargo.toml

```diff
@@ -0,0 +1,15 @@
+[package]
```

Comment:
I searched for `mimalloc` on crates.io... would it simplify things at all if we just took a dependency on https://crates.io/crates/mimalloc  (or one of the many alternatives)?

Reply:
Great call out. We need to customize our compilation of mimalloc, otherwise I would have used it as you suggested.